File: frontend/src/app/sites/seven-wonders/seven-wonders.service.ts
import {Injectable} from '@angular/core';
import {GameConfig} from "./dto/game/GameConfig";
import {GameSessionService} from "../core/http/game-session.service";
import {GameState} from "./dto/game/GameState";
import {environment} from "../../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {NotificationService} from "../../main/core/services/notification.service";

@Injectable({
  providedIn: 'root',
})
export class SevenWondersService extends GameSessionService<GameConfig, GameState> {
  protected override baseUri: string = environment.httpApiUrl + '/jolly';

  constructor(httpClient: HttpClient, notification: NotificationService) {
    super(httpClient, notification);
  }
}

File: backend/src/main/java/net/zorphy/backend/site/sevenwonders/service/SevenWondersService.java
package net.zorphy.backend.site.sevenwonders.service;

import net.zorphy.backend.site.core.http.service.GameSessionService;
import net.zorphy.backend.site.core.http.service.SavableService;
import net.zorphy.backend.site.sevenwonders.dto.game.GameConfig;
import net.zorphy.backend.site.sevenwonders.dto.game.GameState;
import net.zorphy.backend.site.sevenwonders.dto.result.DuelResultState;

public interface SevenWondersService extends GameSessionService<GameConfig, GameState>, SavableService<GameState, DuelResultState> {
}
